src/Voron/Data/Fixed/FixedSizeIterators.cs

```diff
@@ -113,7 +113,7 @@ public long CurrentKey
                 {
                     if (_pos >= _header->NumberOfEntries)
                         throw new InvalidOperationException("Invalid position, cannot read past end of tree");
-                    return FixedSizeTreePage.GetEntry(_dataStart, _pos, _fst._entrySize)->Key;
+                    return FixedSizeTreePage.GetEntry(_dataStart, (int)_pos, _fst._entrySize)->Key;
```

Comment:
Shouldn't we check if `_pos > int.MaxValue` and throw then? Just in case - same as you did above

Reply:
We have a `if (_pos >= _header->NumberOfEntries)` check. Since the `NumberOfEntries` is ushort then I assume that this is not needed.